cpp/src/request_am.cpp

```diff
@@ -293,28 +293,32 @@ void RequestAm::populateDelayedSubmission()
 
   request();
 
-  auto log = [this](const void* buffer, const size_t length) {
+  auto log = [this](const void* buffer, const size_t length, const ucs_memory_type_t memoryType) {
     if (_enablePythonFuture)
-      ucxx_trace_req_f(
-        _ownerString.c_str(),
-        _request,
-        _operationName.c_str(),
-        "buffer %p, size %lu, future %p, future handle %p, populateDelayedSubmission",
-        buffer,
-        length,
-        _future.get(),
-        _future->getHandle());
+      ucxx_trace_req_f(_ownerString.c_str(),
+                       _request,
+                       _operationName.c_str(),
+                       "buffer %p, size %lu, memoryType: %lu, future %p, future handle %p, "
+                       "populateDelayedSubmission",
+                       buffer,
+                       length,
+                       memoryType,
+                       _future.get(),
+                       _future->getHandle());
     else
       ucxx_trace_req_f(_ownerString.c_str(),
                        _request,
                        _operationName.c_str(),
-                       "buffer %p, size %lu, populateDelayedSubmission",
+                       "buffer %p, size %lu, memoryType: %lu, populateDelayedSubmission",
                        buffer,
-                       length);
+                       length,
+                       memoryType);
   };
 
   std::visit(data::dispatch{
-               [this, &log](data::AmSend amSend) { log(amSend._buffer, amSend._length); },
+               [this, &log](data::AmSend amSend) {
```

Comment:
`data::AmSend const &amSend` ?

I guess these objects are small enough that they're right on the cusp of the "pass-by-value" vs. "pass by const reference" boundary.

Reply:
That's right, they're all small enough to be pass-by-value. I think it would be cleaner for us to follow the same practice everywhere (when permitted), either pass-by-value or pass-by-reference, so it seems like we should be passing by reference. WDYT?

If we choose pass-by-reference, then we may want to leave https://github.com/rapidsai/ucxx/pull/121#discussion_r1422408110 as it is currently.